From 4e799e611455663c94ef36e7378b93f4eb72f740 Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Tue, 8 Dec 2015 18:51:56 -0800
Subject: [PATCH 087/173] Snap: Separate default focus time between
 camera/video

For the camera, the user expects a default tap-to-focus duration of 3 seconds
with exposure locking. For the camcorder, however, the user expects a tap-to-focus
duration of 0 seconds with no exposure locking, as the rapid change in exposure
while recording would negatively impact the video.

Separate the tap-to-focus option to reflect the recommended defaults.

[mikeioannina]: Adapted to Snap

Change-Id: I75bc0610f5d301c9728b4bc104b86e0fab5493fb
Signed-off-by: Sultanxda <sultanxda@gmail.com>
---
 res/values/cm_strings.xml               | 1 +
 res/xml/video_preferences.xml           | 2 +-
 src/com/android/camera/VideoModule.java | 2 +-
 3 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/res/values/cm_strings.xml b/res/values/cm_strings.xml
index 5b8ddf306..1f7992239 100644
--- a/res/values/cm_strings.xml
+++ b/res/values/cm_strings.xml
@@ -111,6 +111,7 @@
     <!-- Touch focus duration -->
     <string name="pref_camera_focustime_title">Touch focus duration</string>
     <string name="pref_camera_focustime_default" translatable="false">3000</string>
+    <string name="pref_camera_video_focustime_default" translatable="false">200</string>
 
     <string name="pref_camera_focustime_entry_0">0s</string>
     <string name="pref_camera_focustime_entry_3">3s</string>
diff --git a/res/xml/video_preferences.xml b/res/xml/video_preferences.xml
index 56bcfcbc7..da84cae1a 100644
--- a/res/xml/video_preferences.xml
+++ b/res/xml/video_preferences.xml
@@ -191,7 +191,7 @@
             camera:entryValues="@array/pref_video_focusmode_entryvalues" />
     <ListPreference
             camera:key="pref_camera_video_focustime_key"
-            camera:defaultValue="@string/pref_camera_focustime_default"
+            camera:defaultValue="@string/pref_camera_video_focustime_default"
             camera:title="@string/pref_camera_focustime_title"
             camera:entries="@array/pref_camera_focustime_duration_entries"
             camera:entryValues="@array/pref_camera_focustime_duration_values" />
diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index d91d361af..02246d229 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -2841,7 +2841,7 @@ public class VideoModule implements CameraModule,
         // Set focus time.
         mFocusManager.setFocusTime(Integer.decode(
                 mPreferences.getString(CameraSettings.KEY_VIDEOCAMERA_FOCUS_TIME,
-                mActivity.getString(R.string.pref_camera_focustime_default))));
+                mActivity.getString(R.string.pref_camera_video_focustime_default))));
 
         // Set face detetction parameter.
         String faceDetection = mPreferences.getString(
-- 
2.17.1

